### DISCUSSION TITLE
Animations declared in code behind don't work

### DISCUSSION THREAD

**Discussion by Mopy18** [AvaloniaUI/Avalonia#16480] [Q&A]
### Describe the bug

I tried to implement [this example](https://docs.avaloniaui.net/docs/guides/graphics-and-animation/keyframe-animations#running-animation-from-the-code-behind) from the documentation, but with the animation being declared in code behind instead of xaml. When I declare the animation in xaml like in the example it works as intended, but when it is declared in code behind nothing happens.

### To Reproduce

So basically my UI has a Rectangle and a button, when the button is clicked it creates the animation and runs it:

```
var animation = new Animation();
animation.Duration = TimeSpan.FromSeconds(3);

animation.Children.Add(new KeyFrame
{
    Cue = new Cue(0.0),
    Setters =
    {
        new Setter { Property = Rectangle.OpacityProperty, Value = 0 },
    }
});

animation.Children.Add(new KeyFrame
{
    Cue = new Cue(1.0),
    Setters =
    {
        new Setter { Property = Rectangle.OpacityProperty, Value = 1 },
    }
});

await animation.RunAsync(Rect);
```

Nothing happens when I click the button, the opacity doesn't change at all. Note that if I run this code instead:
```
var animation = (Animation)this.Resources["ResourceAnimation"];
await animation.RunAsync(Rect);
```

it works as intended ("ResourceAnimation" being declared in xaml as done in the example). 

I also tried animating different properties, using KeyTime instead of Cue, playing with Duration, PlaybackDirection, etc. Still nothing happens.

### Avalonia version

11.1.0

### OS

Windows, macOS

**Comment by maxkatz6** [comment]
OpacityProperty is a doable property, while you set an integer.
XAML compiler automatically converts this value, but C# is not.

Change `Value = 0` and `Value = 1` to `Value = 0d` and `Value = 1d`.

**Reply by maxkatz6** [reply]
Also, when running your code, I got InvalidCastException, which quickly helped me to find this issue. You can consider enabling all exceptions by default in IDE.

**Reply by Mopy18** [reply]
I feel dumb... Thank you for your time and answer!
